docs/architecture/director-component.md

```diff
@@ -0,0 +1,42 @@
+# Director Component
+
+## Overview
+
+This document describes [Director](../terminology.md)'s architecture.
+
+## Architecture
+
+The following diagram illustrates the overall architecture of this application:
+
+![](./assets/director-architecture.svg)
+
+### Root Resolver
+
+Root Resolver is a type, which composes of Domain Resolver types. In a result, Root Resolver consists of all GraphQL resolvers needed to run the GraphQL server.  
+
+### Domain Resolver
+
+Domain Resolver is a set of GraphQL resolvers related to a specific domain. For example, Runtime Domain Resolver contains all GraphQL resolvers related to Runtime type, including queries, mutations and field resolvers.
```

Comment:
I would rephrase this. Currently, on the diagram, there is no Domain Resolver or Runtime Domain Resolver. I would say that Runtime Resolver is one of Domain Resolver types.

Reply:
To be honest firstly I wanted to create "generic" diagram, with Domain 1 Resolver, Domain 1 Service etc. What do you think - would it be better?

Reply:
That may be a better approach as our components are not a fixed set.

Reply:
@crabtree What about now?

Reply:
👍